src/main/java/com/iexec/worker/pubsub/StompClientService.java

```diff
@@ -250,6 +268,7 @@ public void afterConnected(StompSession session, StompHeaders connectedHeaders)
             stompSession = session;
             // notify subscribers
             eventPublisher.publishEvent(new SessionCreatedEvent());
+            sessionReady.countDown();
```

Comment:
Why not using exiting + new event :
```
eventPublisher.publishEvent(new SessionCreatedEvent());

eventPublisher.publishEvent(new SessionLostEvent());
```
With something like a lock, updated by events, in subscription service?

Reply:
I don't think a `Lock` would work. We need an object that can be waited for, without the waiter acquiring the lock on it. The only sync object I know that fulfills this constraint is a `CountDownLatch`. This object should also have the ability to be reset. Regarding that point... Nothing exists in Java core.
But I'd like your event-oriented idea, i'll have a look at it.

Reply:
So, I went with a `Condition` object. Let me know if it suits you.